### ISSUE TITLE
[Enhancement]: Require minimum client version when looking up servers

### ISSUE THREAD

**Issue by jonbarrow** [PretendoNetwork/account#114]
### Checked Existing

- [X] I have checked the repository for duplicate issues.

### What enhancement would you like to see?

When looking up server details, for example in NEX token requests, no client version check is performed. This allows any version of any title to be used, even if that version does not match the latest official update.

This has the possibility to create instances where incompatible clients are using the same servers. It also means that cases like the security patches for `ENLBufferPwn` in Mario Kart 7 are not being enforced.

**Comment by BDMCGaming** [comment]
I was going to take a look into trying to implement this through a title list csv, but I'm noticing a bit of an issue. I've been to at least tell that it's using hexadecimal for both X-Nintendo-Application-Version (NNAS) and gamever (NASC). The number is throws doesn't seem to be very consistent with the version numbers from Kinnay's and/or DaniElectra's list. 

Some games like MH3U (and its packet relay tool) and Splatoon seem to add an extra 16 compared to the version list. Some games like Super Smash Bros 4 (Wii U) and Minecraft actually match up with the title list. Mario Kart 8 seems to arbitarily add a 1 to the version on top of the 16. It's all just very inconsistent around games.

And then there are 3DS titles, which have super high version numbers. Despite this, games seem to return very low version numbers through NASC that don't translate anywhere close to what Dani's list has (values of 0001 and 0022 for games in the thousands). 

Is there anything that can be done to get an accurate version list that matches what games are sending, because it seems that version numbers are all over the place to do anything with?

**Comment by jonbarrow** [comment]
I'm not sure what you're talking about tbh. What you're describing doesn't match what I've seen at all. The `X-Nintendo-Application-Version` is the latest title version, hex encoded and left-padded to 4 characters. I'm not sure what version list you're looking at or where you're seeing these extra numbers though

I just checked several dumps and nothing you've described lines up. For example, the version sent by Splatoon is `0120`, or 288 in decimal. This matches the latest version of the Splatoon update title as reported by [tagaya](https://nintendo.wiki/wiki/Online/Nintendo_Network/eShop/Tagaya):

<img width="590" height="160" alt="Image" src="https://github.com/user-attachments/assets/1fea7cf0-cdc7-45fd-817c-52e18403a211" />
<img width="220" height="79" alt="Image" src="https://github.com/user-attachments/assets/b263192b-5bd0-48d6-9ad6-c38798cdafef" />

**Comment by BDMCGaming** [comment]
The output of Tagaya matches the versions numbers I'm seeing. I'm looking at https://danielectra.github.io/kinnay-title-list/view.html?page=wiiu though, as I don't have a dump of Tagaya's output.

**Comment by jonbarrow** [comment]
You said Splatoon adds "16" to the number though, that's the part I'm not understanding. Where are you seeing these extra numbers?

**Comment by BDMCGaming** [comment]
The latest version of Splatoon is v272 according to the title list. It is not v288 as the Application Version would say.

<img width="817" height="23" alt="Image" src="https://github.com/user-attachments/assets/7e446044-55e0-486d-940a-1c28d37122d1" />

**Comment by jonbarrow** [comment]
OH, I thought you meant you were seeing something like `012016` or `16288` or something

**Comment by BDMCGaming** [comment]
Yeah. I'm assuming the version numbers from Tagaya are not the same as the numbers on the title list, which is where I'm running into issues

**Comment by jonbarrow** [comment]
If you were still interested in doing this, here's a list of the latest title versions from Tagaya [latest-versions.json](https://github.com/user-attachments/files/31577900/latest-versions.json)

The limitation here is that this doesn't include any system titles and such, Tagaya doesn't report that
